Completion

Kankur0 opened the pull request:
I've completed the assignment and added the two files necessary to have the application function properly. The Readme also contains documentation on how to create and modify this application. 

This application takes in a JSON object and outputs the values contained within each item within that JSON object in proper html format based on the values for each item.

This application can accept as many items as needed and will output the majority of the most common html elements used for forms..

Kankur0 committed:
Updated Readme.
Created function to convert a JSON array object into valid html
Created an array.json file with some sample data to be converted with app.js

Kankur0 committed:
Fixed markdown in readme.

gmoore10 commented on app.js:
```diff
@@ -0,0 +1,102 @@
+            });
+            
+            //Create loop to convert to html
+            items.forEach(function() {
```
Since this is a forEach, the first parameter of function() can be made available as the iterator like so:

`items.forEach(function(item) {`

This will remove the need for line 28 (instantiation of j) and line 86 (increment of j) as the forEach function is already taking care of that work.

Then line 40 can be changed from:

`items[j]`
to
`item`
...and the forEach will still operate as expected.

Kankur0 replied:
I actually didn't know about that, thank you for letting me know. I use counters a lot, and am always looking for ways to condense my code, so this information will help me a lot in the future!

reactivepixel commented on app.js:
```diff
@@ -0,0 +1,102 @@
+const http = require('http');
+
+const hostname = '127.0.0.1';
```
Don't hard code ip addresses into your codebase. This should be an environmental variable.

reactivepixel commented on app.js:
```diff
@@ -0,0 +1,102 @@
+const http = require('http');
+
+const hostname = '127.0.0.1';
+const port = 3000;
```
Don't hard code the port if you can help it. Creating a default fallback with an environmental var is a better approach.

``` javascript
const port = process.env.port || 3000;
```

gmoore10 commented:
Hi Toby,

Your project looks very good overall.

I added one comment above reflecting a change to the forEach loop implementation which should simplify the code slightly.

Thanks,

Garrett

Kankur0 committed:
Implemented the change suggested by Garrett Moore

Kankur0 committed:
Added technologies to follow DWA Style Guide
Changed code to pass DWA Style Guide check

Kankur0 committed:
Fixed mistake with _eslint.js file
Updated Readme to include Style Guide Info and Author

Kankur0 commented:
I've added more technologies to this pull request to ensure that all code follows DWA Style Guide.

gmoore10 commented on README.md:
```diff
@@ -34,4 +34,52 @@ This application is reusable and will convert any number of items into valid htm
+  * `npm install chai`
+* [ESLint](https://eslint.org/docs/user-guide/getting-started) - Installed locally
+  * Install ESLint
+  `npm install esling --save-dev`
```
I think this might be a typo.

`npm install eslint --save-dev` is what I believe it was supposed to say.

Kankur0 replied:
Good catch! I'll fix that right away.

gmoore10 commented:
Everything looks good overall. Only one typo in the readme found.

I would also recommend adding a `.gitignore` file to the root of your repo with the following line:

`node_modules/`

This will tell git to completely ignore the node_modules folder when committing. A developer can then just run `npm install` to restore the folder in its entirety.

Kankur0 committed:
Fixed typo in Readme
Added gitignore file as suggested by Garrett Moore

gmoore10 commented:
The line in your readme for launching your app `node app.js` needs to be adjusted as you moved your source files into the app folder. It should instead say `node app/app.js`

Kankur0 committed:
Updated Readme to show the proper command to run the application since files were moved.

gmoore10 reviewed (approved):
All fixes applied. Approved.

Kankur0 committed:
Removed Node Modules

reactivepixel commented on app/app.js:
```diff
@@ -0,0 +1,96 @@
+const http = require('http');
+
+const hostname = '127.0.0.1';
```
Best Practice: Hard coding server addresses and ports limits the portability of your code. This should be handled with an environmental variable, and default to 127... if the envar is not present.

reactivepixel commented on app/array.json:
```diff
@@ -0,0 +1 @@
+[{"type":"tEXt"},{"type":"select"},{"type":"button"},{"type":"form"},{"type":"label"},{"type":"email"}, {"type":"button"}]
```
Requirements: Your logic is slightly flawed in that your example you are not accounting for the id or name fields of forms, which are a key points of how forms work in html.

reactivepixel commented on app/app.js:
```diff
@@ -0,0 +1,96 @@
+        let element = '';
+
+        //  Create switch statement for values
+        switch (item) {
```
Requirements: The logical implementation you have built strips out any possible configuration within your array.json file and replaces it with empty form tags.

reactivepixel commented on README.md:
```diff
@@ -1 +1,85 @@
+  `./node_modules/.bin/eslint --init`
+  * Copy contents from [Max Bot's eslint file](https://github.com/reactivepixel/Max-Bot/blob/dev/test/__eslint.js) to your own __eslint.js_ file
+    * Change the _'bot'_ to match the path(s) to your files.
+  * Ensure that your local .eslintrc.json file has the following configurations
```
If you include the .eslintrc.json file in the PR you can expect it to be there and this note of documentation is redundant.

Kankur0 committed:
Performed recommended changed by Mr. Chapman
- Removed hardcoded port and hostname and replace with env variable.
- Removed unnecessary documentation from readme.
- Accounted for more complex json data
- - Create two functions to check for attributes and text
- - Reformatted switch statement to account for these two new functions

Kankur0 committed:
Removed unnecessary code
